components/builder-originsrv/tests/data_store/mod.rs

```diff
@@ -1714,6 +1714,104 @@ fn get_origin_channel() {
 }
 
 #[test]
+fn promote_origin_package_group() {
+    let ds = datastore_test!(DataStore);
+    let mut origin = originsrv::OriginCreate::new();
+    origin.set_name(String::from("core"));
+    origin.set_owner_id(1);
+    origin.set_owner_name(String::from("scottkelly"));
+    ds.create_origin(&origin).expect("Should create origin");
+
+    let neurosis = ds.get_origin_by_name("core")
+        .expect("Could not retrieve origin")
+        .expect("Origin does not exist");
+
+    // Create a new origin channel
+    let mut oscc = originsrv::OriginChannelCreate::new();
+    oscc.set_origin_id(neurosis.get_id());
+    oscc.set_origin_name(neurosis.get_name().to_string());
+    oscc.set_name(String::from("eve"));
+    oscc.set_owner_id(1);
+    let channel1 = ds.create_origin_channel(&oscc).expect(
+        "Failed to create origin channel",
+    );
+
+    let mut ident1 = originsrv::OriginPackageIdent::new();
+    ident1.set_origin("core".to_string());
+    ident1.set_name("cacerts".to_string());
+    ident1.set_version("2017.01.17".to_string());
+    ident1.set_release("20170209064044".to_string());
```

Comment:
You have a good point, but this test code reads like all of the other test code in this file.  At some point, it might be worth refactoring all of these tests to use helper functions and clean up the verbosity, but as it is, someone reading through this test code today won't have any trouble following my test because it's like all of the other tests. I fear that choosing this moment to refactor one test will be jarring and make this test harder to follow.